test/cpp/api/optim.cpp

```diff
@@ -168,7 +168,7 @@ TEST(OptimTest, OptimizerAccessors) {
 TEST(OptimTest, BasicInterface) {
   struct MyOptimizer : Optimizer {
     using Optimizer::Optimizer;
-    void step() override {}
+    torch::Tensor step(LossClosure closure = nullptr) override { return {};}
   };
   std::vector<torch::Tensor> parameters = {
       torch::ones({2, 3}), torch::zeros({2, 3}), torch::rand({2, 3})};
```

Comment:
It seems that we actually don't have the equivalent `XORConvergence_LBFGS` and `ProducesPyTorchValues_LBFGS` tests like the other optimizers. We can have this as a follow-up PR after the branch cut (if before is not enough time), and before the final release.